src/InlineEntityFormControllerInterface.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\inline_entity_form\InlineEntityFormControllerInterface.
+ */
+
+namespace Drupal\inline_entity_form;
+
+use Drupal\Component\Plugin\ConfigurablePluginInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Defines the interface for entity browser widgets.
+ */
+interface InlineEntityFormControllerInterface extends ConfigurablePluginInterface {
+
+  /**
+   * Returns an array of css filepaths for the current entity type, keyed
+   * by theme name.
+   *
+   * If provided, the "base" CSS file is included for all themes.
+   * If a CSS file matching the current theme exists, it will also be included.
+   *
+   * @code
+   * return [
+   *   'base' => drupal_get_path('module', 'test_module') . '/css/inline_entity_form.base.css',
+   *   'seven' => drupal_get_path('module', 'test_module') . '/css/inline_entity_form.seven.css',
+   * ];
+   * @endcode
+   *
+   * @return array
+   *   List of css files for inclusion keyed by theme name.
+   */
+  public function css();
+
+  /**
+   * Returns an array of entity type labels (singular, plural) fit to be
+   * included in the UI text.
+   *
+   * @return array
+   *   Array containing two values:
+   *     - singular: label for singular form,
+   *     - plural: label for plural form.
+   */
+  public function labels();
+
+  /**
+   * Returns an array of fields used to represent an entity in the IEF table.
+   *
+   * The fields can be either Field API fields or properties defined through
+   * hook_entity_property_info().
+   *
+   * Modules can alter the output of this method through
+   * hook_inline_entity_form_table_fields_alter().
+   *
+   * @param $bundles
+   *   An array of allowed bundles for this widget.
+   *
+   * @return
+   *   An array of field information, keyed by field name. Allowed keys:
+   *   - type: 'field' or 'property',
+   *   - label: Human readable name of the field, shown to the user.
+   *   - weight: The position of the field relative to other fields.
+   *   Special keys for type 'field', all optional:
+   *   - formatter: The formatter used to display the field, or "hidden".
+   *   - settings: An array passed to the formatter. If empty, defaults are used.
+   *   - delta: If provided, limits the field to just the specified delta.
+   */
+  public function tableFields($bundles);
+
+  /**
+   * Returns a setting value.
+   *
+   * @param $name
+   *   The name of the setting value to return.
+   *
+   * @return
+   *   A setting value.
+   */
+  public function getSetting($name);
+
+  /**
+   * Returns the entity type managed by this controller.
+   *
+   * @return
+   *   The entity type.
+   */
+  public function entityType();
```

Comment:
EntityType or EntityTypeId? I am fine with both but we have to clarify. Maybe similar to EntityInterface?